### PR TITLE
docs: add readme

### PR DESCRIPTION
**Pull request by GCdePaula**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,78 @@
...
+## What's in a name
+
+Our fraud-proof system is called _Dave_.
+Just Dave.
+It's neither an acronym nor an abbreviation, but a name.
+Like most names, it should be written in lower case with an initial capital, that is, "Dave".
+
+Dave is permissionless.
+This means anyone can participate in the consensus.
+Since anyone can participate, there's the possibility of Sybil attacks, where an attacker can generate an army of fake personas and try to shift the consensus to their favour.
```
Comment (guidanoli) [inline README.md:25]: *in their favor?

File: README.md [hunk 2]
```diff
@@ -1,3 +1,78 @@
...
+
+Dave's security is one of N: a single honest validator can enforce the correct result.
+It doesn't matter if it's you against the world.
+If you're honest, Dave's got your back; you can fight a mountain of powerful, well-funded crooks and win, using a single laptop in a timely manner.
+
+Dave is named after the David vs. Goliath archetype.
```
Comment (guidanoli) [inline README.md:31]: *inspired by the David vs. Goliath story?

File: README.md [hunk 3]
```diff
@@ -1,3 +1,78 @@
...
+The former implements the RV64GC ISA, while the later implements the much smaller RV64IM ISA.
+Using a technique called _machine swapping_ and leveraging good compilers, we implement in Solidity only the micro-architecture's state-transition function, while the execution environment can support a much larger set of extensions.
+
+Nevertheless, Dave was designed to be agnostic on its execution environment.
+As long as one can provide a self-contained state-transition function, Dave will work.
+
+
+## Algorithm
+
+Dave is based over the Permissionless Refereed Tournaments primitive.
```
Comment (guidanoli) [inline README.md:49]: *based on?

File: README.md [hunk 4]
```diff
@@ -1,3 +1,78 @@
...
+## Contributing
+
+Thank you for your interest in Cartesi!
+Head over to our [Contributing Guidelines](CONTRIBUTING.md) for instructions on how to sign our Contributors Agreement and get started with Cartesi!
+
+Please note we have a [Code of Conduct](CODE_OF_CONDUCT.md), please follow it in all your interactions with the project.
+
+## License
+
+The repository and all contributions are licensed under [APACHE 2.0](https://www.apache.org/licenses/LICENSE-2.0).
```
Comment (guidanoli) [inline README.md:69]: *Apache?

**Review by guidanoli — COMMENTED** [review]
offchain :arrow_right: off-chain
onchain :arrow_right: on-chain

**Comment by claudioantonio** [comment]
It good to me, after giving Guilherme's suggestions. 😍

File: README.md (hunk 2 above)
Comment (GCdePaula) [reply README.md:31]: Better! But I'll keep "archetype".

File: README.md (hunk 4 above)
Comment (GCdePaula) [reply README.md:69]: Our current readmes write it in upper case (_e.g._ noether).

**Comment by GCdePaula** [comment]
Can you try to follow the instructions to run the node?

@claudioantonio @guidanoli @miltonjonat

File: README.md [hunk 5]
```diff
@@ -1,3 +1,78 @@
...
+Dave's security is one of N: a single honest validator can enforce the correct result.
+It doesn't matter if it's you against the world.
+If you're honest, Dave's got your back; you can fight a mountain of powerful, well-funded crooks and win, using a single laptop in a timely manner.
+
+Dave is inspired by the David vs. Goliath archetype.
+
+
+## Execution Environment
+
+Dave uses the [Cartesi machine](https://github.com/cartesi/machine-emulator) as its execution environment.
```
Comment (miltonjonat) [inline README.md:36]: I'd use "Cartesi Machine" with a capital M here and on other parts. I don't understand why you sometimes capitalize it and sometimes not

File: README.md [hunk 6]
```diff
@@ -1,3 +1,78 @@
...
+The Cartesi machine is a RISC-V emulator.
+Its onchain implementation can be found [here](https://github.com/cartesi/machine-solidity-step).
+The Cartesi Machine state-transition function is implemented in two layers: the big-machine and the micro-architecture.
+The former implements the RV64GC ISA, while the later implements the much smaller RV64IM ISA.
```
Comment (miltonjonat) [inline README.md:40]:
```suggestion
The former implements the RV64GC ISA, while the latter implements the much smaller RV64IM ISA.
```

**Review by miltonjonat — COMMENTED** [review]
Great to have this!

**Comment by miltonjonat** [comment]
> Can you try to follow the instructions to run the node?
> 
> @claudioantonio @guidanoli @miltonjonat

Didn't work for me:

```
 => [ 9/15] ADD ./permissionless-arbitration/contracts/src/ ./src/                                                                                                                                                  0.1s
 => [10/15] RUN forge --version                                                                                                                                                                                     0.3s
 => ERROR [11/15] RUN forge build                                                                                                                                                                                   0.5s
------
 > [11/15] RUN forge build:
0.498 Error: 
0.498 failed to resolve file: "/app/contracts/../../machine-emulator-sdk/solidity-step/ready_src/UArchStep.sol": No such file or directory (os error 2); check configured remappings.
0.498     --> "/app/contracts/src/tournament/abstracts/LeafTournament.sol"
0.498         "step/ready_src/UArchStep.sol"
```

**Comment by miltonjonat** [comment]
Ok, now it worked! I needed to download a bunch of submodules - and that's not on the README.
```
git submodule update --init --recursive
```
(I admit I don't like submodules very much, but that's a personal thing)

**Comment by guidanoli** [comment]
I had to run the command thrice, but maybe that's my internet fault.
Also, I had to run `git submodule update --init --recursive` as well.
This command should be on the Lua node README.

**Comment by GCdePaula** [comment]
Nice catches! I've updated with the requests. @claudioantonio can you take a look?

**Comment by claudioantonio** [comment]
It worked beautifully @GCdePaula ! ❤️ 
I only missed a brief description of what is the example, what I executed. 😉

**Comment by miltonjonat** [comment]
Yeah a quick description of what is executed would be cool, I was a bit lost too!

**Comment by guidanoli** [comment]
I agree. I had no idea what was being printed to me, as a newcomer.

**Comment by GCdePaula** [comment]
I've added more info to the readme!

I think we can improve the logging itself later in a different PR. It's good for debugging, but as a prototype showing newcomers what's going on, it may be improved.

Let me know if there's more improvements to the readme.

**Review by miltonjonat — APPROVED** [review]
LGTM

**Comment by augustoteixeira** [comment]
This is looking great!

**Pull request merged**